src/topology/uniform_space/basic.lean

```diff
@@ -2,38 +2,121 @@
 Copyright (c) 2017 Johannes Hölzl. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Johannes Hölzl, Mario Carneiro, Patrick Massot
-
-Theory of uniform spaces.
+-/
+import order.filter.lift
+import topology.separation
+/-!
+# Uniform spaces
 
 Uniform spaces are a generalization of metric spaces and topological groups. Many concepts directly
 generalize to uniform spaces, e.g.
 
-* completeness
-* extension of uniform continuous functions to complete spaces
-* uniform contiunuity & embedding
-* totally bounded
-* totally bounded ∧ complete → compact
-
-The central concept of uniform spaces is its uniformity: a filter relating two elements of the
-space. This filter is reflexive, symmetric and transitive. So a set (i.e. a relation) in this filter
-represents a 'distance': it is reflexive, symmetric and the uniformity contains a set for which the
-`triangular` rule holds.
-
-The formalization is mostly based on the books:
-  N. Bourbaki: General Topology
-  I. M. James: Topologies and Uniformities
-A major difference is that this formalization is heavily based on the filter library.
+* uniform continuity (in this file)
+* completeness (in `cauchy.lean`)
+* extension of uniform continuous functions to complete spaces (in `uniform_embedding.lean`)
+* totally bounded sets (in `cauchy.lean`)
+* totally bounded complete sets are compact (in `cauchy.lean`)
+
+A uniform structure on a type `X` is a filter `𝓤 X` on `X × X` satisfying some conditions
+which makes it reasonable to say that `∀ᶠ (p : X × X) in 𝓤 X, ...` means
+"for all p.1 and p.2 in X close enough, ...". Elements of this filter are called entourages
+of `X`. The two main examples are:
+
+* If `X` is a metric space, `V ∈ 𝓤 X ↔ ∃ ε > 0, { p | dist p.1 p.2 < ε } ⊆ V`
+* If `G` is an additive topological group, `V ∈ 𝓤 G ↔ ∃ U ∈ 𝓝 (0 : G), {p | p.2 - p.1 ∈ U} ⊆ V`
+
+Those examples are generalizations in two different directions of the elementary example where
+`X = ℝ` and `V ∈ 𝓤 ℝ ↔ ∃ ε > 0, { p | |p.2 - p.1| < ε } ⊆ V` which features both the topological
+group structure on `ℝ` and its metric space structure.
+
+Each uniform structure on `X` induces a topology on `X` characterized by:
+`nhds_eq_comap_uniformity : ∀ {x : X}, 𝓝 x = comap (prod.mk x) (𝓤 X)`
+where `prod.mk x : X → X × X := (λ y, (x, y))` is the partial evaluation of the product
```

Comment:
```suggestion
Each uniform structure on `X` induces a topology on `X` characterized by
* `nhds_eq_comap_uniformity : ∀ {x : X}, 𝓝 x = comap (prod.mk x) (𝓤 X)`
where `prod.mk x : X → X × X := (λ y, (x, y))` is the partial evaluation of the product
```
Guessing the equation is supposed to be on its own line?

Reply:
Indeed it would be nicer on its own line, but I'm not sure we should use a markdown list here. Maybe a `>` as in
> `nhds_eq_comap_uniformity : ∀ {x : X}, 𝓝 x = comap (prod.mk x) (𝓤 X)`

Reply:
Sure, anything to separate it.

Reply:
Done